src/hooks/useProjectLocationListFilter.ts

```diff
@@ -0,0 +1,55 @@
+import { setDistrictList, setDivisionList, setSubDivisionList } from '@/reducers/listsSlice';
+import { RootState } from '@/store';
+import _ from 'lodash';
+import { useEffect } from 'react';
+import { useAppDispatch, useAppSelector } from './common';
+
+const useProjectLocationListFilter = () => {
+  const project = useAppSelector((state: RootState) => state.project.selectedProject, _.isEqual);
+  const districts = useAppSelector((state: RootState) => state.location.districts, _.isEqual);
+  const divisions = useAppSelector((state: RootState) => state.location.divisions, _.isEqual);
+  const subDivisions = useAppSelector((state: RootState) => state.location.subDivisions, _.isEqual);
+  const districtsList = useAppSelector((state: RootState) => state.lists.district);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    const selectedDistrict = districts.find((d) => d.id === project?.projectLocation);
+
+    if (districtsList.length <= 0) {
+      dispatch(setDistrictList(districts));
+    }
+
+    if (project?.projectLocation) {
+      /* If selected is a district, we need to find all divisions that have that district as their parent */
+      if (selectedDistrict) {
+        dispatch(setDivisionList(divisions.filter((d) => d.parent === project?.projectLocation)));
+      }
+
+      const selectedDivision = divisions.find((d) => d.id === project?.projectLocation);
+      /* If selected is a division, we need to find the district and find all divisions for that district */
+      if (selectedDivision) {
+        const districtForDivision = districts.find((d) => d.id === selectedDivision?.parent);
+
+        dispatch(setDivisionList(divisions.filter((d) => d.parent === districtForDivision?.id)));
```

Comment:
Will setDivisionList set empty list if no match occur in filtering? if yes, then refactor to this:

dispatch(setDivisionList(districtForDivision ? divisions.filter((d) => d.parent === districtForDivision.id) || []));